server/src/main/java/org/opensearch/search/pipeline/Pipeline.java

```diff
@@ -168,6 +199,27 @@ SearchResponse transformResponse(SearchRequest request, SearchResponse response)
         0,
         Collections.emptyList(),
         Collections.emptyList(),
+        Collections.emptyList(),
         null
     );
+
+    <Result extends SearchPhaseResult> SearchPhaseResults<Result> runSearchPhaseTransformer(
+        SearchPhaseResults<Result> searchPhaseResult,
+        SearchPhaseContext context,
+        String currentPhase,
+        String nextPhase
+    ) throws SearchPipelineProcessingException {
+
+        try {
+            for (SearchPhaseResultsProcessor searchPhaseResultsProcessor : searchPhaseResultsProcessors) {
+                if (currentPhase.equals(searchPhaseResultsProcessor.getBeforePhase().getName())
+                    && nextPhase.equals(searchPhaseResultsProcessor.getAfterPhase().getName())) {
+                    searchPhaseResult = searchPhaseResultsProcessor.process(searchPhaseResult, context);
+                }
+            }
+            return searchPhaseResult;
+        } catch (Exception e) {
```

Comment:
Does anything throw a checked exception? If not, better to catch a `RuntimeException` so that a future refactoring doesn't result in swallowing an interrupt or other important checked exception.

Reply:
I was following the convention which was done in the other part of the code for same Class.
here: https://github.com/opensearch-project/OpenSearch/blob/b040d3bd01c46989d28414dbc7694c8079283502/server/src/main/java/org/opensearch/search/pipeline/Pipeline.java#L185-L194

Also we are catching and throwing the exception. So interrupts will not be missed. I think we should be good here.

Reply:
To be interrupt safe you need to do:

```
if (e instance of InterruptedException) {
    Thread.currentThread().interrupt();
}
```

Otherwise the interrupted thread continues on without the interrupt status set and is free to block again. I see that `SearchResponseProcessor.process` declares `throws Exception`. Is that necessary? If not, can we create an issue to change that signature so that consumers don't have to deal with checked exceptions?

Reply:
Github issue: https://github.com/opensearch-project/OpenSearch/issues/8304